app/controllers/participations_controller.rb

```diff
@@ -11,32 +11,42 @@ def index
   def show
   end
 
-  # GET /participations/new
+  # GET /events/1/register
   def new
-    @participation = Participation.new
+    event = Event.find(params[:event_id])
+    if event.event_type.nil?
+      redirect_to root_url, notice: "Ehhez az eseményhez nincs jelentkező űrlap"
+    elsif event.participations.exists?(:user => current_user)
```

Comment:
I think it is better to stick to newer hash synstax insted of the hash rocket, if possible
```suggestion
    elsif event.participations.exists?(user: current_user)
```